geomstats/_backend/numpy/__init__.py

```diff
@@ -381,6 +382,18 @@ def array_from_sparse(indices, data, target_shape):
     return array(coo_matrix((data, list(zip(*indices))), target_shape).todense())
 
 
+def vec_to_diag(vec):
```

Comment:
This function is already present here:
https://github.com/geomstats/geomstats/blob/master/geomstats/algebra_utils.py

As a guideline, let's put algebra utils in this `algebra_utils.py` file, using the `gs.` syntax to write in 4 backends. This avoid re-writing the same function in the 4 backends folders. The `_backend` folder is only here to write **already existing numpy functions** in the other backends, so that we can use the `gs.` syntax following the numpy API. It is not here to put utils from algebra. 

The exceptions are:
- if the implementation of the function cannot be written using the same backends `gs.` functions across the 4 backends,
- if we can get considerable speed-ups by writing the algebra function differently in the 4 backends.

Reply:
@SaitejaUtpala what do you think of this?

Reply:
@ninamiolane  both tensorflow(https://www.tensorflow.org/api_docs/python/tf/linalg/diag) and pytorch (https://pytorch.org/docs/1.8.1/generated/torch.diag_embed.html?highlight=diag#torch.diag_embed) has inbuilt support for this and only  numpy doesn't have this thing

Reply:
we can write two versions in numpy which uses assignment operation and which doesn't

Reply:
so I think this should be in backend itself

Reply:
and this is so common in ML. I think numpy may natively support this in future

Reply:
I will remove from_vector_to_diagonal_matrix from algebra utils in next PR

Reply:
Ok, agreed, this function should be in the backends. You say that numpy does not have this but how about this:
https://numpy.org/doc/stable/reference/generated/numpy.diag.html
?

Reply:
because on 2d array it doesn't return 3d array. (which is what we want)
`If v is a 2-D array, return a copy of its k-th diagonal. If v is a 1-D array, return a 2-D array with v on the k-th diagonal`.

Reply:
Goooot it, thank you for the explanation.